contracts/DxMgnPool.sol

```diff
@@ -124,26 +130,41 @@ contract DxMgnPool is Ownable {
     }
 
     function triggerMGNunlockAndClaimTokens() public {
-        require(currentState() == State.PoolingEnded, "Pooling period is not yet over.");
+        checkForStateUpdate();
+        require(currentState == State.PoolingEnded, "Pooling period is not yet over.");
         require(
             dx.getAuctionIndex(address(depositToken), address(secondaryToken)) > lastParticipatedAuctionIndex, 
             "Last auction is still running"
         );      
-        
+
         (address sellToken, address buyToken) = buyAndSellToken();
         address(dx).call(abi.encodeWithSignature("claimSellerFunds(address,address,address,uint256)", buyToken, sellToken, address(this), lastParticipatedAuctionIndex));
         mgnToken.unlockTokens();
         totalDeposit = dx.balances(address(depositToken), address(this));
         if(totalDeposit > 0){
             dx.withdraw(address(depositToken), totalDeposit);
         }
+        currentState = State.DepositWithdrawnFromDx;
     }
 
+    bool public isMagnoliaWithdrawnFromDX = false;
```

Comment:
Is this still needed? Once we have executed this method, we will be in `State.MgnUnlocked` and thus should not be able to call this method again.

Reply:
thanks for notifying me. Yes, it is not needed at all..